src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp

```diff
@@ -0,0 +1,426 @@
+      gr::Tags::SpatialMetric<3, Frame::Inertial, DTYPE(data)>>                \
+  TovStar<STYPE(data)>::variables(                                             \
+      const tnsr::I<DTYPE(data), 3>& /*x*/,                                    \
+      tmpl::list<                                                              \
+          gr::Tags::SpatialMetric<3, Frame::Inertial, DTYPE(data)>> /*meta*/,  \
+      const RadialVariables<DTYPE(data)>& radial_vars) const noexcept;         \
+  template tuples::TaggedTuple<                                                \
+      typename TovStar<STYPE(data)>::template DerivSpatialMetric<DTYPE(data)>> \
+  TovStar<STYPE(data)>::variables(                                             \
+      const tnsr::I<DTYPE(data), 3>& x,                                        \
+      tmpl::list<typename TovStar<STYPE(                                       \
+          data)>::template DerivSpatialMetric<DTYPE(data)>> /*meta*/,          \
+      const RadialVariables<DTYPE(data)>& radial_vars) const noexcept;         \
+  template tuples::TaggedTuple<gr::Tags::SqrtDetSpatialMetric<DTYPE(data)>>    \
+  TovStar<STYPE(data)>::variables(                                             \
+      const tnsr::I<DTYPE(data), 3>& /*x*/,                                    \
+      tmpl::list<gr::Tags::SqrtDetSpatialMetric<DTYPE(data)>> /*meta*/,        \
+      const RadialVariables<DTYPE(data)>& radial_vars) const noexcept;         \
+  template tuples::TaggedTuple<                                                \
+      gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DTYPE(data)>>         \
+  TovStar<STYPE(data)>::variables(                                             \
+      const tnsr::I<DTYPE(data), 3>& /*x*/,                                    \
+      tmpl::list<gr::Tags::InverseSpatialMetric<3, Frame::Inertial,            \
+                                                DTYPE(data)>> /*meta*/,        \
+      const RadialVariables<DTYPE(data)>& radial_vars) const noexcept;         \
+  template tuples::TaggedTuple<                                                \
+      gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DTYPE(data)>>           \
+  TovStar<STYPE(data)>::variables(                                             \
+      const tnsr::I<DTYPE(data), 3>& /*x*/,                                    \
+      tmpl::list<gr::Tags::ExtrinsicCurvature<3, Frame::Inertial,              \
+                                              DTYPE(data)>> /*meta*/,          \
+      const RadialVariables<DTYPE(data)>& radial_vars) const noexcept;
+
+#define INSTANTIATE(_, data)                                \
+  template class TovStar<STYPE(data)>;                      \
+  template bool operator!=(const TovStar<STYPE(data)>& lhs, \
+                           const TovStar<STYPE(data)>& rhs) noexcept;
+
+GENERATE_INSTANTIATIONS(INSTANTIATE_VARS, (gr::Solutions::TovSolution),
+                        (double, DataVector))
```

Comment:
You should also be able to loop over the tags to shorten this even more

Reply:
deferring this until template parameters of tags are consistent